### PR TITLE
docs: fix broken links in markdown files

### PR DESCRIPTION
**Pull request by rina23q**
## Proposed changes
Fix broken links in our markdown files.

I used [this GitHub action](https://github.com/gaurav-nelson/github-action-markdown-link-check) to find broken links. However, I wound not add this to our PR check as I found many false-positives. You can check the [result](https://github.com/rina23q/thin-edge.io/actions/runs/9663794079/job/26656865897) that I ran it on my fork against the current main branch contents. 

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [x] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: CODING_GUIDELINES.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Coding Guidelines
 
 ## Code Style
-Follow [Rust coding guidelines](https://github.com/rust-dev-tools/fmt-rfcs/blob/master/guide/guide.md).
+Follow [Rust coding guidelines](https://github.com/rust-lang/rust/tree/HEAD/src/doc/style-guide/src).
```
Comment (rina23q) [inline CODING_GUIDELINES.md:4]: I'm 100% sure if this guide is originally intended. Now we can't know what was the original link's content as it's 404 :/

Comment (didier-wenzek) [reply CODING_GUIDELINES.md:4]: I would rather points to the on-line documentation and not source book: https://doc.rust-lang.org/style-guide/index.html

**Review by didier-wenzek — COMMENTED** [review]
Only one point has to be fixed: the link to the rust style book (see: https://github.com/thin-edge/thin-edge.io/pull/2962#discussion_r1653183338).

File: CODING_GUIDELINES.md (hunk 1 above)
Comment (rina23q) [reply CODING_GUIDELINES.md]: fixed

**Review by didier-wenzek — APPROVED** [review]
Approved

**Review by reubenmiller — APPROVED** [review]
Approved. Thanks for the fixes

**Pull request merged**